src/ZkFold/Base/Protocol/Protostar.hs

```diff
@@ -66,7 +68,7 @@ instance (Arithmetic a, KnownNat n) => SpecialSoundProtocol a (RecursiveCircuit
     --
     algebraicMap rc _ _ _ =
         let
-            varF (NewVar ix) = var (ix P.+ value @n)
+            varF (NewVar ix) = var (toConstant (decode @VarField $ fromStrict ix) P.+ value @n)
```

Comment:
This code is obsolete, I migrated to Vars a long time ago. See [my branch](https://github.com/zkFold/zkfold-base/blob/vks4git/protostar/src/ZkFold/Base/Protocol/Protostar/ArithmeticCircuit.hs#L97) for details.
Can you please revert changes in this file so I have less merge conflicts?